Blogplace.Tests.Integration/Tests/CommentsTests.cs

```diff
@@ -0,0 +1,171 @@
+
+        comments.Should().NotBeEmpty();
+        comments.Should().ContainSingle(x =>
+            x.ArticleId == ArticlesRepositoryFake.StandardUserArticle!.Id &&
+            x.Content == CommentsRepositoryFake.StandardUserCommentOnStandardUserArticle!.Content
+        );
+    }
+
+    [Test]
+    public async Task SearchByParent_CommentsShouldBeReturned()
+    {
+        //Arrange
+        var client = this._factory.CreateClient_Standard();
+        var articleId = ArticlesRepositoryFake.StandardUserArticle!.Id;
+        var parentCommentId = CommentsRepositoryFake.StandardUserCommentOnStandardUserArticle!.Id;
+
+        var commentContent = "CHILD_COMMENT_CONTENT";
+        var createChildRequest = new CreateCommentRequest(articleId, commentContent, parentCommentId);
+        var createChildResponse = await client.PostAsync($"{this.urlBaseV1}/Comments/Create", createChildRequest);
+        createChildResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        var request =
+            new SearchCommentsByParentRequest(CommentsRepositoryFake.StandardUserCommentOnStandardUserArticle!.Id);
+
+        //Act
+        var response = await client.PostAsync($"{this.urlBaseV1}/Comments/SearchByParent", request);
+
+        //Assert
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+        var comments = (await response.Content.ReadFromJsonAsync<SearchCommentsResponse>())!.Comments.ToArray();
+
+        comments.Should().NotBeEmpty();
+        comments.Should().ContainSingle(x => x.ArticleId == articleId && x.Content == commentContent);
+    }
+
+    [Test]
+    public async Task Update_CommentShouldBeUpdated()
+    {
+        //Arrange
+        var client = this._factory.CreateClient_Standard();
+        var commentId = await this.CreateComment(
+            client,
+            ArticlesRepositoryFake.StandardUserArticle!.Id,
+            "TEST_CONTENT"
+        );
+
+        var newContent = "UPDATED_CONTENT";
+        var request = new UpdateCommentRequest(
+            commentId,
+            newContent
+        );
+
+        //Act
+        var response = await client.PostAsync($"{this.urlBaseV1}/Comments/Update", request);
+
+        //Assert
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+        // TODO: Check if the comment was updated in the repository
+    }
+
+    [Test]
+    public async Task Delete_CommentShouldBeDeleted()
+    {
+        //Arrange
+        var client = this._factory.CreateClient_Standard();
+        var commentId = await this.CreateComment(
+            client,
+            ArticlesRepositoryFake.StandardUserArticle!.Id,
+            "TEST_COMMENT_CONTENT"
+        );
+        var request = new DeleteCommentRequest(commentId);
+
+        //Act
+        var response = await client.PostAsync($"{this.urlBaseV1}/Comments/Delete", request);
+
+        //Assert
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+        // TODO: Check if the comment doesn't exist in the repository
```

Comment:
Tak jak komentarz wyżej, zamiast sprawdzania repo, można sprawdzać odpowiedź z api

Usuwanie niekoniecznie musi być fizycznym usunięciem z bazy, lepiej sprawdzać czy po usunięciu nie pojawia sie na liście

Można to zaprojektować na różne sposoby, testy powinny być odporne na zmiany szczegółów w kodzie